objects/src/mock/chain.rs

```diff
@@ -55,3 +55,401 @@ pub fn mock_chain_data(consumed_notes: &mut [Note]) -> ChainMmr {
 
     chain_mmr
 }
+
+#[cfg(feature = "mock")]
+mod mock {
+    use crate::{
+        assets::Asset,
+        builder::{
+            AccountBuilder, AccountIdBuilder, FungibleAssetBuilder, NonFungibleAssetBuilder,
+            DEFAULT_ACCOUNT_CODE,
+        },
+        notes::{Note, NoteInclusionProof, NOTE_LEAF_DEPTH, NOTE_TREE_DEPTH},
+        Account, AccountError, AccountId, AccountType, AssetError, BlockHeader, ChainMmr, Felt,
+        StorageItem, Vec,
+    };
+    use crypto::{
+        hash::rpo::RpoDigest as Digest,
+        merkle::{DefaultMerkleStore, MerkleError, NodeIndex, SimpleSmt, TieredSmt},
+    };
+    use miden_core::{FieldElement, StarkField};
+    use rand::{Rng, SeedableRng};
+
+    /// Initial timestamp value
+    const TIMESTAMP_START: Felt = Felt::new(1693348223);
+    /// Timestep of timestamp on each new block
+    const TIMESTAMP_STEP: Felt = Felt::new(10);
+
+    #[derive(Default, Debug, Clone)]
+    #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+    pub struct Objects<R: Rng> {
+        accounts: Vec<Account>,
+        fungible_faucets: Vec<(AccountId, FungibleAssetBuilder)>,
+        nonfungible_faucets: Vec<(AccountId, NonFungibleAssetBuilder<R>)>,
+        notes: Vec<Note>,
+        nullifiers: Vec<Digest>,
+    }
+
+    impl<R: Rng> Objects<R> {
+        pub fn new() -> Self {
+            Self {
+                accounts: vec![],
+                fungible_faucets: vec![],
+                nonfungible_faucets: vec![],
```

Comment:
Is the idea to store all historical states here as well? If we don't need to store the historical states, then these could be `BTreeMap`'s.

Reply:
I was adding the historical state to the merkle store, so this would have a single entry per item.

I didn't see an advantage to using a map, but we can change it if you want. What key do you want to use for it?

Reply:
I was thinking the key would be account ID and this way we naturally guarantee that we can't have more than one account with the same ID in the set.

Reply:
how would someone use the account then? if the data is created in a binary, and we only have the JSON file to load the data from, the `AccountId` is long gone. Iterating over the keys would not give the same order as the one used in during creation, so people would need to copy/paste the id meaning the data creating a dependency among them.

Reply:
> Wouldn't the full BTreeMap be serialized in the JSON file? And the order should be the same as the BTreeMap would be sorted by the numeric value of account ID.

The account id is effectively a random value, so that means there is a total order, but that it is a total order of random values. If the initial chain has an account created with initial assets, and the tests wants to use that, it would need to search for the account instead of using the creation order.